File: source/bindings/js/src/sdk/TranslationSynthesisEventArgs.ts
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT license.
import { SessionEventArgs, TranslationSynthesisResult } from "./Exports";

/**
 * Translation Synthesis event arguments
 * @class TranslationSynthesisEventArgs
 */
export class TranslationSynthesisEventArgs extends SessionEventArgs {
    private privResult: TranslationSynthesisResult;

    /**
     * Creates and initializes an instance of this class.
     * @constructor
     * @param {TranslationSynthesisResult} result - The translation synthesis result.
     * @param {string} sessionId - The session id.
     */
    public constructor(result: TranslationSynthesisResult, sessionId?: string) {
        super(sessionId);

        this.privResult = result;
    }

    /**
     * Specifies the translation synthesis result.
     * @member TranslationSynthesisEventArgs.prototype.result
     * @function
     * @public
     * @returns {TranslationSynthesisResult} Specifies the translation synthesis result.
     */
    public get result(): TranslationSynthesisResult {
        return this.privResult;
    }
}

File: source/bindings/java/com/microsoft/cognitiveservices/speech/translation/TranslationSynthesisEventArgs.java
//
// Copyright (c) Microsoft. All rights reserved.
// Licensed under the MIT license. See LICENSE.md file in the project root for full license information.
//
package com.microsoft.cognitiveservices.speech.translation;

import com.microsoft.cognitiveservices.speech.SessionEventArgs;
import com.microsoft.cognitiveservices.speech.util.Contracts;
import com.microsoft.cognitiveservices.speech.util.IntRef;
import com.microsoft.cognitiveservices.speech.util.SafeHandle;

/**
 * Defines the content of translation synthesis result event.
 */
public final class TranslationSynthesisEventArgs extends SessionEventArgs
{
    /*! \cond INTERNAL */

    /**
     * Constructs an instance of a TranslationRecognitionEventArgs object.
     * @param eventArgs recognition event args object.
     */
    TranslationSynthesisEventArgs(long eventArgs) {
        super(eventArgs);
        storeEventData(false);

    }
    
    TranslationSynthesisEventArgs(long eventArgs, boolean dispose) {
        super(eventArgs);
        storeEventData(dispose);
    }

    /*! \endcond */

    /**
     * Specifies the translation synthesis result.
     * @return Specifies the translation synthesis result.
     */
    public final TranslationSynthesisResult getResult() {
        return result;
    }

    /**
     * Returns a String that represents the translation recognition result event.
     * @return A String that represents the translation recognition result event.
     */
    @Override
    public String toString()
    {
        return "SessionId:" + this.getSessionId() + " Result:" + result.toString()  + ".";
    }

    private void storeEventData(boolean disposeNativeResources) {
        Contracts.throwIfNull(eventHandle, "eventHandle");
        IntRef resultHandle = new IntRef(0);
        Contracts.throwIfFail(getSynthesisResult(eventHandle, resultHandle));
        this.result = new TranslationSynthesisResult(resultHandle.getValue());
        Contracts.throwIfNull(this.getSessionId(), "SessionId");
        if (disposeNativeResources == true) {
            super.close();
        }
    }
    
    private TranslationSynthesisResult result;

    private final native long getSynthesisResult(SafeHandle eventHandle, IntRef resultHandle);
}
